src/Cli/dotnet/commands/dotnet-help/HelpCommand.cs

```diff
@@ -99,18 +92,18 @@ public int Execute()
                 Reporter.Error.WriteLine(
                     string.Format(
                         LocalizableStrings.CommandDoesNotExist,
-                        _parseResult.GetValue(HelpCommandParser.Argument)).Red());
+                        helpArgs).Red());
                 Reporter.Output.WriteLine(HelpUsageText.UsageText);
                 return 1;
             }
         }
 
-        private bool TryGetDocsLink(string commandName, out string docsLink)
+        private bool TryGetDocsLink(string[] command, out string docsLink)
         {
-            var command = Parser.GetBuiltInCommand(commandName);
-            if (command != null && command as DocumentedCommand != null)
+            var parsedCommand = Parser.Instance.Parse(["dotnet", .. command]);
+            if (parsedCommand != null && parsedCommand.CommandResult is not null && parsedCommand.CommandResult.Command is DocumentedCommand dc)
```

Comment:
nit: Simplification that handles all the null checks in one.
```suggestion
            if (parsedCommand?.CommandResult?.Command is DocumentedCommand dc)
```